### PR TITLE
Upgrade buildtools to xunit 2.1.0 Release

### PR DESCRIPTION
**Pull request by ianhays**
This PR and a yet-to-be-posted CoreFX PR accompanying it will update to the latest xunit.

### CODE REVIEW

File: src/Microsoft.DotNet.Build.Tasks/Targets/test-runtime/project.json [hunk 1]
```diff
@@ -8,12 +8,12 @@
...
-      "Microsoft.DotNet.xunit.performance.analysis": "1.0.0-alpha-build0014",
-      "Microsoft.DotNet.xunit.performance.runner.Windows": "1.0.0-alpha-build0014",
+      "Microsoft.DotNet.xunit.performance.analysis": "1.0.0-alpha-build0022",
+      "Microsoft.DotNet.xunit.performance.runner.Windows": "1.0.0-alpha-build0022",
 
-      "xunit": "2.1.0-rc1-build3168",
-      "xunit.console.netcore": "1.0.2-prerelease-00088",
-      "xunit.runner.utility": "2.1.0-rc1-build3168"
+      "xunit": "2.1.0",
+      "xunit.console.netcore": "1.0.2-prerelease-00100",
```
Comment (ianhays) [inline src/Microsoft.DotNet.Build.Tasks/Targets/test-runtime/project.json]: Might need to modify this. I want the project.json to reference the xunit.console.netcore package that will be built when this PR merges.

File: src/Microsoft.DotNet.Build.Tasks/Targets/test-runtime/project.lock.json [hunk 2]
```diff
@@ -1878,33 +1878,7 @@
           "lib/dotnet/xunit.assert.dll": {}
         }
       },
-      "xunit.console.netcore/1.0.2-prerelease-00088": {
```
Comment (ianhays) [inline src/Microsoft.DotNet.Build.Tasks/Targets/test-runtime/project.lock.json]: Not sure why this reference got removed from the project.lock.json completely. Should be referencing version 100.

To test the build of BuildTools, I used a localpublish of xunit,console.netcore with the version set to 100 to the buildtools/packages folder, then rebuilt and localpublished everything to the CoreFX repo for testing.

**Comment by ianhays** [comment]
@mellinoe @weshaggard

**Comment by mellinoe** [comment]
LGTM. I'm fine if you need to just do another commit right after in order to get the project.lock.json files to agree with each other, as well.

**Comment by ianhays** [comment]
Thanks @mellinoe. It looks like that's what I'm going to have to do. I'll merge this, publish a new package, then submit another PR with the updated lock files.

File: src/Microsoft.DotNet.Build.Tasks/Targets/test-runtime/project.lock.json (hunk 2 above)
Comment (weshaggard) [inline src/Microsoft.DotNet.Build.Tasks/Targets/test-runtime/project.lock.json]: Yes you should try and understand what happen here as I agree it doesn't look right.

**Comment by weshaggard** [comment]
LGTM. Are you planning to update all the test projects in corefx to release as well?

**Comment by ianhays** [comment]
@weshaggard  Yup. That's in https://github.com/dotnet/corefx/pull/3599/

**Pull request merged**